File: psra/model/src/main/scala/by/arhor/psra/repository/PhotoRepository.scala
package by.arhor.psra.repository

import java.util

import by.arhor.psra.model.Photo
import org.springframework.data.mongodb.repository.{MongoRepository, Query}

trait PhotoRepository
  extends MongoRepository[Photo, String]
    with BasicRepository[Photo, String] {

  @Query("{ 'enabled' : true, 'tags' : { $in : ?0 } }")
  def findByAnyOfTags(tags: Array[String]): util.List[Photo]

  @Query("{ 'enabled' : true, 'tags' : { $all : ?0 } }")
  def findByAllOfTags(tags: Array[String]): util.List[Photo]

}

File: psra/model/src/main/java/by/arhor/psra/repository/PhotoRepository.java
package by.arhor.psra.repository;

import java.util.List;

import by.arhor.psra.model.Photo;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;


public interface PhotoRepository
  extends MongoRepository<Photo, String>
        , BasicRepository<Photo, String> {

  @Query("{ 'enabled' : true, 'tags' : { $in : ?0 } }")
  List<Photo> findByAnyOfTags(String[] tags);

  @Query("{ 'enabled' : true, 'tags' : { $all : ?0 } }")
  List<Photo> findByAllOfTags(String[] tags);

}
